### DISCUSSION TITLE
Is Creating Multiple Globals an Anti-Pattern?

### DISCUSSION THREAD

**Discussion by Bradleykingz** [payloadcms/payload#2916] [General]
I'm in the process of setting up a multi-tenancy enabled application. Every user should have their own "space" such that no documents ever overlap. There's only one problem - globals are meant to be for "one-off" components, such that no two instances of the same component should ever exist. Keep in mind, however, that every user should have their own navbar, and their own footer, "breaking" the concept of what a global is.

Are there perhaps ways I can work around this? For instance, creating globals on a per-user basis using hooks? And, if so, are there any ways things might break which aren't immediately obvious?

**Comment by Bradleykingz** [comment]
Since this is a "special case", I decided to use Mongoose directly and create a new global for every tenant that's created.

**Reply by TechSynthesis** [reply]
Hey @Bradleykingz How do you create multiple globals, I've been trying to figure this out

**Reply by IAmNatch** [reply]
We solved this issue in another discussion: https://github.com/payloadcms/payload/discussions/3268

**Comment by IAmNatch** [comment]
Would be curious if there's an official way to do this, now that multi-tenancy is being promoted more within payload.
